bitbots_localization/include/bitbots_localization/ObservationModel.h

```diff
@@ -0,0 +1,117 @@
+//
+// Created by judith on 09.03.19.
+//
+
+#ifndef BITBOTS_LOCALIZATION_OBSERVATIONMODEL_H
+#define BITBOTS_LOCALIZATION_OBSERVATIONMODEL_H
+
+#include <particle_filter/ParticleFilter.h>
+#include <bitbots_localization/RobotState.h>
+#include <bitbots_localization/map.h>
+#include <bitbots_localization/tools.h>
+#include <humanoid_league_msgs/LineInformationRelative.h>
+#include <humanoid_league_msgs/GoalRelative.h>
+#include <humanoid_league_msgs/PixelsRelative.h>
+#include <humanoid_league_msgs/FieldBoundaryRelative.h>
+
+namespace hlm = humanoid_league_msgs;
+
+class RobotPoseObservationModel : public particle_filter::ObservationModel<RobotState> {
+
+public:
+
+    /**
+     * empty
+     */
+    RobotPoseObservationModel(std::shared_ptr<Map> map_lines, std::shared_ptr<Map> map_goals,
+                              std::shared_ptr<Map> map_field_boundary, std::shared_ptr<Map> map_corners,
+                              std::shared_ptr<Map> map_t_crossings, std::shared_ptr<Map> map_crosses);
+
+    /**
+     * empty
+     */
+    ~RobotPoseObservationModel();
+
+    /**
+     *
+     * @param state Reference to the state that has to be weightened.
+     * @return weight for the given state.
+     */
+    double measure(const RobotState &state) const;
+
+    void set_measurement_lines(hlm::LineInformationRelative measurement);
+
+    void set_measurement_goal(hlm::GoalRelative measurement);
+
+    void set_measurement_field_boundary(hlm::FieldBoundaryRelative measurement);
+
+    void set_measurement_corners(hlm::PixelsRelative measurement);
+
+    void set_measurement_t_crossings(hlm::PixelsRelative measurement);
+
+    void set_measurement_crosses(hlm::PixelsRelative measurement);
+
+    std::vector<std::pair<double, double>> get_measurement_lines() const;
+
+    std::vector<std::pair<double, double>> get_measurement_goals() const;
+
+    std::vector<std::pair<double, double>> get_measurement_field_boundary() const;
+
+    std::vector<std::pair<double, double>> get_measurement_corners() const;
+
+    std::vector<std::pair<double, double>> get_measurement_t_crossings() const;
+
+    std::vector<std::pair<double, double>> get_measurement_crosses() const;
```

Comment:
Suggest to use a trailing return type for this functions `auto` instead of `std::vector<std::pair<double, double>>`.

Reply:
We prefer not to use `auto` when the type helps the readability of the interface.